lib/recipes-data/src/lib/utils.ts

```diff
@@ -37,7 +37,26 @@ export function nowTime():Date {
   return new Date();
 }
 
-export const extractCropIdFromGuimUrl = (url: string): string | undefined =>
-	url.match(
-		/https:\/\/.*?\/(?<id>.*?)\/(?<cropId>\d{1,4}_\d{1,4}_\d{1,4}_\d{1,4})\/(?<fileName>.*?)$/,
-	)?.groups?.cropId;
+export const extractCropDataFromGuimUrl = (
+	url: string,
+): { mediaId: string; cropId: string; width: number } | undefined => {
+	const match = url.match(
+		/https:\/\/.*?\/(?<mediaId>.*?)\/(?<cropId>\d{1,4}_\d{1,4}_(?<width>\d{1,4})_\d{1,4})\/(?<fileName>.*?)$/,
```

Comment:
I may be wrong (_reading_ regexes is hard!) but is the `<fileName>` group needed? Not a blocker, just thought I'd ask :)

Reply:
It's not! I included it to illustrate the shape of the URL, if that makes sense. If it's unnecessary or distracting, we can remove.